packages/zapp/console/src/components/Executions/ExecutionDetails/ExecutionTabContent.tsx

```diff
@@ -0,0 +1,174 @@
+    makeNodeExecutionDynamicWorkflowQuery(dynamicParents),
+  );
+  const [initializeNodes, setInitilizeNodes] = useState<dNode[]>([]);
+
+  useEffect(() => {
+    const nodes: dNode[] = compiledWorkflowClosure
+      ? transformerWorkflowToDag(compiledWorkflowClosure, dynamicWorkflows).dag.nodes
+      : [];
+    // we remove start/end node info in the root dNode list during first assignment
+    const initializeNodes = convertToPlainNodes(nodes);
+    setInitilizeNodes(initializeNodes);
+  }, [compiledWorkflowClosure, dynamicWorkflows]);
+
+  const [selectedNodes, setSelectedNodes] = useState<string[]>([]);
+
+  // Note: flytegraph allows multiple selection, but we only support showing
+  // a single item in the details panel
+  const [selectedExecution, setSelectedExecution] = useState<NodeExecutionIdentifier | null>(
+    selectedNodes.length
+      ? nodeExecutionsById[selectedNodes[0]]
+        ? nodeExecutionsById[selectedNodes[0]].id
+        : {
+            nodeId: selectedNodes[0],
+            executionId: nodeExecutionsById[Object.keys(nodeExecutionsById)[0]].id.executionId,
+          }
+      : null,
+  );
+
+  const [selectedPhase, setSelectedPhase] = useState<TaskExecutionPhase | undefined>(undefined);
+  const [isDetailsTabClosed, setIsDetailsTabClosed] = useState<boolean>(!selectedExecution);
+
+  useEffect(() => {
+    setIsDetailsTabClosed(!selectedExecution);
+  }, [selectedExecution]);
+
+  const onCloseDetailsPanel = () => {
+    setSelectedExecution(null);
+    setSelectedPhase(undefined);
+    setSelectedNodes([]);
+  };
+
+  const [chartTimezone, setChartTimezone] = useState(TimeZone.Local);
+
+  const handleTimezoneChange = (tz) => setChartTimezone(tz);
+
+  const timelineContext = useMemo(
+    () => ({ selectedExecution, setSelectedExecution }),
+    [selectedExecution, setSelectedExecution],
+  );
+
+  const onNodeSelectionChanged = (newSelection: string[]) => {
+    const validSelection = newSelection.filter((nodeId) => {
+      if (nodeId === startNodeId || nodeId === endNodeId) {
+        return false;
+      }
+      return true;
+    });
+    setSelectedNodes(validSelection);
+    const newSelectedExecution = validSelection.length
+      ? nodeExecutionsById[validSelection[0]]
+        ? nodeExecutionsById[validSelection[0]].id
+        : {
+            nodeId: validSelection[0],
+            executionId: nodeExecutionsById[Object.keys(nodeExecutionsById)[0]].id.executionId,
+          }
+      : null;
+    setSelectedExecution(newSelectedExecution);
+  };
+
+  const renderContent = () => {
+    switch (tabType) {
+      case tabs.timeline.id:
+        return (
+          <div className={styles.wrapper}>
+            <div className={styles.container}>
+              <NodeExecutionsTimelineContext.Provider value={timelineContext}>
+                <ExecutionTimeline
+                  chartTimezone={chartTimezone}
+                  initializeNodes={initializeNodes}
+                />
+                ;
```

Comment:
looks like this semicolon doesn't belong here 😄

Reply:
fixed